ai16z/eliza
# feat: Updated characters types, Discord & Telegram enhancements

# Relates to:

[Issue 399](https://github.com/ai16z/eliza/issues/399)




# Risks

Low: Additional settings to character.json types are low level changes, and should not break any core functionality. Additionally discord messages.ts and telegram file additions are specific to the new settings, so risk is low.

# Background

## What does this PR do?

Enhances the Discord and Telegram bot's message handling to support:

1. Configurable mentions-only mode via shouldRespondOnlyToMentions setting
2. Allows bot joins to restricted telegram groups via shouldOnlyJoinInAllowedGroups setting

## What kind of change is this?

Features (non-breaking change which adds functionality)
Improvements (misc. changes to existing features)

# Documentation changes needed?

My changes require a change to the project documentation.

Add documentation for new shouldRespondOnlyToMentions config option
Add documentation for new shouldOnlyJoinInAllowedGroups config option

# Testing

## Where should a reviewer start?

1. Review changes in messages.ts (Discord) and telegramClient.ts (Telegram), particularly:

- New configuration option handling

## Detailed testing steps

1. Test single agent mentions-only mode:

- Set shouldRespondOnlyToMentions: true
- Verify agent only responds to direct mentions
- Verify no responses to non-mention messages

2. Test group add limitation for Telegram:

- Set shouldOnlyJoinInAllowedGroups: true
- Add TG bot to any chat with an empty array for approvedGroups and observer agent leave
- Add TG bot to any chat with an array for approvedGroups and observer agent stay in chat


## ninja_dev - Discord Username
